packages/firebase-auth-rn/lib/index.js

```diff
@@ -13,7 +13,7 @@ GoogleSignin.configure({webClientId: '711003129445-2js3agurj1cg8vleo0s4o4reppep4
 const signInWithCredential = async (credential, email) => {
   const [provider] = await auth().fetchSignInMethodsForEmail(email)
 
-  if (credential.providerId === provider) {
+  if (!provider || credential.providerId === provider) {
```

Comment:
this if statement is very confusing. It's hard to tell what it really does. Move it a separate function with clear name that indicates what it does